api/handlers_test.go

```diff
@@ -255,6 +256,28 @@ func TestGetByIDSessionHandlerFunc(t *testing.T) {
 			})
 		})
 	})
+
+	Convey("Given a valid request", t, func() {
+		mockCache := &CacheMock{
+			GetByIDFunc: func(ID string) (*session.Session, error) {
+				return nil, SessionExpired
```

Comment:
I would add a test for when the error returned from GetByID is an unknown error then a 500 internal service error is expected in the response status code.

I understand the cache package only returns two types of errors but the handler shouldn't have to know this level of detail and we should have a test that proves it can handle the scenario. Also it will force you to think about my previous comment as this test will show you are setting a response code of 0 instead of 500.

for example:
```
...
	mockCache := &CacheMock{
		GetByIDFunc: func(ID string) (*session.Session, error) {
			return nil, errors.New("unexpected error")
		},
	}
...
```

Reply:
Added the new test